esiclient/v1/orchestrator/openshift.py

```diff
@@ -0,0 +1,368 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import json
+import logging
+import os
+import requests
+import time
+
+from osc_lib.command import command
+from osc_lib.i18n import _
+
+from esiclient import utils
+
+
+BASE_ASSISTED_INSTALLER_URL = \
+    'https://api.openshift.com/api/assisted-install/v2/'
+
+
+def call_assisted_installer_api(url, method, headers={}, data=None):
+    full_url = BASE_ASSISTED_INSTALLER_URL + url
+    if method == 'post':
+        response = requests.post(full_url, headers=headers, json=data)
+    elif method == 'get':
+        response = requests.get(full_url, headers=headers)
+    elif method == 'patch':
+        response = requests.patch(full_url, headers=headers, json=data)
+    else:
+        raise OSAIException(
+            'Unknown method used when calling Assisted Installer')
+    if response.status_code not in [200, 201, 202, 204]:
+        raise OSAIException(
+            "Unexpected response from Assisted Installer (%s): %s"
+            % (response.status_code, response.reason))
+    return response
+
+
+def wait_for_nodes(infra_env_id, headers, num_nodes, target_status):
```

Comment:
Do you think adding a timeout might be useful? Sometimes nodes may have unexpected issues during booting?